src/stream/src/executor/wrapper/stream_node_metrics.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright 2025 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use futures_async_stream::try_stream;
+use tokio::time::Instant;
+
+use crate::executor::monitor::ProfileMetricsExt;
+use crate::executor::prelude::*;
+
+#[try_stream(ok = Message, error = StreamExecutorError)]
+pub async fn stream_node_metrics(
+    enable_explain_analyze_stats: bool,
```

Comment:
Notice how we use `enable_explain_analyze_stats` before this commit: https://github.com/risingwavelabs/risingwave/pull/20715/commits/879c5ad61d90fd6b3bd088e92930e17b51d9e51d.

We had the following structure:
```rs
if enable_explain_analyze_stats {
    #[for_await]
  for m in stream { 
    yield m?; 
  }
} else {
    #[for_await]
  for m in stream { 
    // do profiling
    yield m?; 
  }
}
```

But this leads to stack overflow. I think it prevents some inlining optimizations from happening, and the result is that the generated state machine struct for `stream_node_metrics` stream is really large (maybe x2?). Hence I changed it here to avoid the branch.

Reply:
```
  #[for_await]
  for m in stream { 
    // do profiling _conditionally_
    yield m?;
  }
```

Reply:
No more stack overflow is encountered.

Reply:
(I used lldb to pinpoint it to this exact function call)

Reply:
Would you try `Box::pin` first? This should help.

Reply:
I think it will move the problem from stack to heap, and consume more memory there. So I try to avoid it with this solution.